test/integration/confirmations/signatures/permit.test.tsx

```diff
@@ -0,0 +1,239 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { ApprovalType } from '@metamask/controller-utils';
+import mockMetaMaskState from '../../data/integration-init-state.json';
+import { integrationTestRender } from '../../../lib/render-helpers';
+import { shortenAddress } from '../../../../ui/helpers/utils/util';
+import * as backgroundConnection from '../../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventLocation,
+} from '../../../../shared/constants/metametrics';
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+import { createMockImplementation } from '../../helpers';
+
+jest.mock('../../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBackgroundConnection = jest.mocked(backgroundConnection);
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const getMetaMaskStateWithUnapprovedPersonalSign = (accountAddress: string) => {
```

Comment:
Rename `getMetaMaskStateWithUnapprovedPersonalSign` to `getMetaMaskStateWithUnapprovedPermitSignature`

Reply:
nice catch, copy past error 🙈  changed